code/__DEFINES/_globals.dm

```diff
@@ -58,3 +58,12 @@
 
 //Create an typed null global
 #define GLOBAL_DATUM(X, Typepath) GLOBAL_RAW(Typepath/##X); GLOBAL_UNMANAGED(X)
+
+GLOBAL_LIST_EMPTY(string_files)
+
+//Create a global to be populated with a file during init
+#define GLOBAL_LIST_FILE_STRING(X, Filename) GLOBAL_LIST(X); \
+/datum/controller/global_vars/InitGlobal##X(){\
+	..();\
+	GLOB.string_files[#X] = Filename;\
```

Comment:
This "different kind of InitGlobal" that doesn't actually init that global but instead sets a key so it will be initted later seems... odd to me, though I guess it works.

Reply:
protected does this i just copied that